src/build/content/prerendered.ts

```diff
@@ -97,7 +121,10 @@ const buildAppCacheValue = async (
   if (
     !meta.status &&
     rsc.includes('NEXT_NOT_FOUND') &&
-    !meta.headers['x-next-cache-tags'].includes('/@')
+    !(
+      typeof meta.headers?.['x-next-cache-tags'] === 'string' &&
+      meta.headers?.['x-next-cache-tags'].includes('/@')
+    )
```

Comment:
Note here - I typed `meta` (see line 80) above, and because `headers` is typed as `OutgoingHttpHeaders` (and not `Record<string,string>`) it required some additional type guard here to satisfy type checking